[1.1.1] pytest fails

When trying to run the tests during the build of the Debian package, I get the error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/lib/python3/dist-packages/pytest/__main__.py", line 5, in <module>
    raise SystemExit(pytest.console_main())
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 192, in console_main
    code = main()
           ^^^^^^
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 150, in main
    config = _prepareconfig(args, plugins)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 331, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/pluggy/_callers.py", line 130, in _multicall
    teardown[0].send(outcome)
  File "/usr/lib/python3/dist-packages/_pytest/helpconfig.py", line 104, in pytest_cmdline_parse
    config: Config = outcome.get_result()
                     ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/usr/lib/python3/dist-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 1075, in pytest_cmdline_parse
    self.parse(args)
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 1433, in parse
    self.args, self.args_source = self._decide_args(
                                  ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 1272, in _decide_args
    self.issue_config_time_warning(
  File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 1465, in issue_config_time_warning
    warnings.warn(warning, stacklevel=stacklevel)
pytest.PytestConfigWarning: No files were found in testpaths; consider removing or adjusting your testpaths configuration. Searching recursively from the current directory instead.
```
I guess that for some reason the tests are not found; however I can't find the cause. Any ideas?
Version 1.1.0 works with otherwise identical packages.
* Python 3.11 and Python 3.12
* Pytest 7.4.3
* Pluggy 1.3.0

Removing the duplicated quotes, at least the tests are executed. However, I now see one failing test:
```
__________________________ test_only_pyproject[True] ___________________________

tmpdir = local('/tmp/pytest-of-pbuilder/pytest-1/test_only_pyproject_True_0')
pyproject_use_helpers = True

    @pytest.mark.parametrize("pyproject_use_helpers", [None, False, True])
    def test_only_pyproject(tmpdir, pyproject_use_helpers):
        """…"""
    
        pytest.importorskip("setuptools", minversion="62.0")
    
        package_name = "helpers_test_package_" + str(uuid.uuid4()).replace("-", "_")
    
        test_pkg = tmpdir.mkdir("test_pkg")
        test_pkg.mkdir(package_name).ensure("__init__.py")
    
        simple_pyx = test_pkg.join(package_name, "simple.pyx")
        simple_pyx.write(
            dedent(
                """\
            def test():
                pass
        """
            )
        )
    
        if pyproject_use_helpers is None:
            extension_helpers_option = ""
        else:
            extension_helpers_option = dedent(
                f"""
            [tool.extension-helpers]
            use_extension_helpers = {str(pyproject_use_helpers).lower()}
            """
            )
    
        test_pkg.join("pyproject.toml").write(
            dedent(
                f"""\
                [project]
                name = "{package_name}"
                version = "0.1"
    
                [tool.setuptools.packages]
                find = {{namespaces = false}}
    
                [build-system]
                requires = ["setuptools>=43.0.0",
                            "wheel",
                            "cython"]
                build-backend = 'setuptools.build_meta'
    
                """
            )
            + extension_helpers_option
        )
    
        install_temp = test_pkg.mkdir("install_temp")
    
        with test_pkg.as_cwd():
            # NOTE: we disable build isolation as we need to pick up the current
            # developer version of extension-helpers
            subprocess.call(
                [
                    sys.executable,
                    "-m",
                    "pip",
                    "install",
                    ".",
                    "--no-build-isolation",
                    f"--target={install_temp}",
                ]
            )
    
        if "" in sys.path:
            sys.path.remove("")
    
        sys.path.insert(0, "")
    
        with install_temp.as_cwd():
>           importlib.import_module(package_name)

extension_helpers/tests/test_setup_helpers.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'helpers_test_package_1d2feb3b_4297_4281_af34_875eec8d7a98'
import_ = <function _gcd_import at 0x7f8303cb7d80>

>   ???
E   ModuleNotFoundError: No module named 'helpers_test_package_1d2feb3b_4297_4281_af34_875eec8d7a98'

<frozen importlib._bootstrap>:1140: ModuleNotFoundError
```
with both Python 3.11 and 3.12.

Interesting. I see that this test indeed fails when I run it locally with `pytest extension_helpers/tests/test_setup_helpers.py`, although with a different error:

<details> <summary> pytest log </summary>

```
======================================== test session starts =========================================
platform darwin -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0
rootdir: /Users/clm/dev/astropy-project/org/extension-helpers
configfile: pyproject.toml
collected 15 items                                                                                   

extension_helpers/tests/test_setup_helpers.py ..............F                                  [100%]

============================================== FAILURES ==============================================
_____________________________________ test_only_pyproject[True] ______________________________________

tmpdir = local('/private/var/folders/47/t9k4zdz96vzd4v1rxyv07nvh0000gn/T/pytest-of-clm/pytest-2/test_only_pyproject_True_0')
pyproject_use_helpers = True

    @pytest.mark.parametrize("pyproject_use_helpers", [None, False, True])
    def test_only_pyproject(tmpdir, pyproject_use_helpers):
        """
        Test that makes sure that extension-helpers can be enabled without a
        setup.py and without a setup.cfg file.
        """
    
        pytest.importorskip("setuptools", minversion="62.0")
    
        package_name = "helpers_test_package_" + str(uuid.uuid4()).replace("-", "_")
    
        test_pkg = tmpdir.mkdir("test_pkg")
        test_pkg.mkdir(package_name).ensure("__init__.py")
    
        simple_pyx = test_pkg.join(package_name, "simple.pyx")
        simple_pyx.write(
            dedent(
                """\
            def test():
                pass
        """
            )
        )
    
        if pyproject_use_helpers is None:
            extension_helpers_option = ""
        else:
            extension_helpers_option = dedent(
                f"""
            [tool.extension-helpers]
            use_extension_helpers = {str(pyproject_use_helpers).lower()}
            """
            )
    
        test_pkg.join("pyproject.toml").write(
            dedent(
                f"""\
                [project]
                name = "{package_name}"
                version = "0.1"
    
                [tool.setuptools.packages]
                find = {{namespaces = false}}
    
                [build-system]
                requires = ["setuptools>=43.0.0",
                            "wheel",
                            "cython"]
                build-backend = 'setuptools.build_meta'
    
                """
            )
            + extension_helpers_option
        )
    
        install_temp = test_pkg.mkdir("install_temp")
    
        with test_pkg.as_cwd():
            # NOTE: we disable build isolation as we need to pick up the current
            # developer version of extension-helpers
            subprocess.call(
                [
                    sys.executable,
                    "-m",
                    "pip",
                    "install",
                    ".",
                    "--no-build-isolation",
                    f"--target={install_temp}",
                ]
            )
    
        if "" in sys.path:
            sys.path.remove("")
    
        sys.path.insert(0, "")
    
        with install_temp.as_cwd():
>           importlib.import_module(package_name)

extension_helpers/tests/test_setup_helpers.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/clm/.pyenv/versions/3.12.0/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1381: in _gcd_import
    ???
<frozen importlib._bootstrap>:1354: in _find_and_load
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb'
import_ = <function _gcd_import at 0x100b7c0e0>

>   ???
E   ModuleNotFoundError: No module named 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb'

<frozen importlib._bootstrap>:1318: ModuleNotFoundError
---------------------------------------- Captured stdout call ----------------------------------------
Processing /private/var/folders/47/t9k4zdz96vzd4v1rxyv07nvh0000gn/T/pytest-of-clm/pytest-2/test_only_pyproject_True_0/test_pkg
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb
  Building wheel for helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb (pyproject.toml): started
  Building wheel for helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb (pyproject.toml): finished with status 'error'
Failed to build helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb
---------------------------------------- Captured stderr call ----------------------------------------
  error: subprocess-exited-with-error
  
  × Building wheel for helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-14.1-arm64-cpython-312
      creating build/lib.macosx-14.1-arm64-cpython-312/helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb
      copying helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb/__init__.py -> build/lib.macosx-14.1-arm64-cpython-312/helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb
      running egg_info
      creating helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info
      writing helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/PKG-INFO
      writing dependency_links to helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/dependency_links.txt
      writing top-level names to helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/top_level.txt
      writing manifest file 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/SOURCES.txt'
      reading manifest file 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/SOURCES.txt'
      writing manifest file 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.egg-info/SOURCES.txt'
      copying helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb/_compiler.c -> build/lib.macosx-14.1-arm64-cpython-312/helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb
      running build_ext
      building 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb.simple' extension
      creating build/temp.macosx-14.1-arm64-cpython-312
      creating build/temp.macosx-14.1-arm64-cpython-312/helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -I/Users/clm/.pyenv/versions/extension-helpers.dev/include -I/Users/clm/.pyenv/versions/3.12.0/include/python3.12 -c ./helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb/simple.c -o build/temp.macosx-14.1-arm64-cpython-312/./helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb/simple.o
      clang: error: no such file or directory: './helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb/simple.c'
      clang: error: no input files
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb
ERROR: Could not build wheels for helpers-test-package-202f1a36-5b70-4d22-a505-13b33c8c02bb, which is required to install pyproject.toml-based projects
====================================== short test summary info =======================================
FAILED extension_helpers/tests/test_setup_helpers.py::test_only_pyproject[True] - ModuleNotFoundError: No module named 'helpers_test_package_202f1a36_5b70_4d22_a505_13b33c8c02bb'
==================================== 1 failed, 14 passed in 8.77s ====================================

```

</details>

However it passes locally when ran through `tox -e py312-test`, which is closer to what happens in CI !

--- a/extension_helpers/tests/test_setup_helpers.py
+++ b/extension_helpers/tests/test_setup_helpers.py
@@ -378,6 +378,7 @@ def test_only_pyproject(tmpdir, pyproject_use_helpers):
         """
         )
 
+    buildtime_requirements = ["setuptools>=43.0.0", "wheel", "Cython"]
     test_pkg.join("pyproject.toml").write(
         dedent(
             f"""\
@@ -389,9 +390,7 @@ def test_only_pyproject(tmpdir, pyproject_use_helpers):
             find = {{namespaces = false}}
 
             [build-system]
-            requires = ["setuptools>=43.0.0",
-                        "wheel",
-                        "cython"]
+            requires = [{', '.join(f'"{_}"' for _ in buildtime_requirements)}]
             build-backend = 'setuptools.build_meta'
 
             """
@@ -404,17 +403,28 @@ def test_only_pyproject(tmpdir, pyproject_use_helpers):
     with test_pkg.as_cwd():
         # NOTE: we disable build isolation as we need to pick up the current
         # developer version of extension-helpers
-        subprocess.call(
-            [
-                sys.executable,
-                "-m",
-                "pip",
-                "install",
-                ".",
-                "--no-build-isolation",
-                f"--target={install_temp}",
-            ]
-        )
+        # In order to do so, we need to ensure that build-time dependencies are
+        # installed first
+        cmd1 = [
+            sys.executable,
+            "-m",
+            "pip",
+            "install",
+            *buildtime_requirements,
+            f"--target={install_temp}",
+        ]
+        subprocess.call(cmd1)
+
+        cmd2 = [
+            sys.executable,
+            "-m",
+            "pip",
+            "install",
+            ".",
+            "--no-build-isolation",
+            f"--target={install_temp}",
+        ]
+        subprocess.call(cmd2)
 
     if "" in sys.path:
         sys.path.remove("")

